### ISSUE TITLE
Is there differences in .NET runtime between Azure Function and App Service?

### ISSUE THREAD

**Issue by tripheo0412** [Azure/Azure-Functions#2150]
Function App name: vahterus-cpq-azure-function
My project is a .NET 6 API backend. It also using an external DLL (sizing.dll) written in Visual Basic .NET, right now, we are hosting it in App Service Windows container, using `mcr.microsoft.com/dotnet/aspnet:6.0.1-windowsservercore-ltsc2019` (we use it because 1 dependencies exproting excel needs gdiplus.dll which only available in windows server core).

We plan on migrating it to azure function .NET 6 windows.

The Azure Function migrated code. In my development environment with .NET 6 SDK. If I ran debugger mode. It runs flawlessly.
Also if I hosted in App Service .NET 6 runtime (or .NET core 3.1) it also runs without hiccup.

But if I deployed to Azure Function v4 (both in process or isolated mode) the external DLL (sizing.dll) written in Visual Basic .NET will not run properly.

So my question is: Is there differences in .NET runtime between Azure Function and App Service?

**Comment by Ved2806** [comment]
Thank you for the feedback! We will check for the possibilities internally and update you with the findings.

**Comment by tripheo0412** [comment]
Figured that my external DLL is targeting .NET framework 4.8.
Switch to Azure Function v1 and it runs fine.
Any idea why it can run on App Service using .NET core 3.1 or 6.0 without hiccup?

**Comment by Ved2806** [comment]
we are investigating this further and will update you soon with the findings

**Comment by Ved2806** [comment]
Hi @fabiocav @brettsam Could you please look into this?

**Comment by kshyju** [comment]
@tripheo0412 What specific error are you getting when deploying the function app to azure? Based on the information you provided, I tried a repro and it worked fine for me when deployed to azure. Here is what I did.

1.  Created a VB class library with .NET Framework 4.8 as target framework. The class library exposes a simple public method.
2.  Built the class library in release mode, thus producing a DLL from the class library.
3.  Created a C# .NET 6 isolated function app(V4), added the above DLL as a project reference (used the "browse" option to pick the DLL).
4. Updated the function code to use the public method from the VB class library.
5. Ran locally & verified it is working. 
6. Used Visual Studio publish option to publish this function app to Azure (Windows consumption plan) and verified it is working.

Are you doing something different than the above?
